vassal-app/src/main/java/VASSAL/build/module/PrivateMap.java

```diff
@@ -116,7 +116,7 @@ else if (SIDE.equals(key)) {
       owners = (String[]) value;
     }
     else if (USE_BOARDS.equals(key)) {
-      if (value == null || ((String) value).isEmpty()) {
+      if (value == null || (value instanceof String && ((String) value).isEmpty())) {
```

Comment:
Maybe I'm over-thinking this, but if `value` *isn't* a String, isn't that going to cause problems with the `else` statement, too?  In which case perhaps this should be `if (value == null || !(value instanceof String) || ((String) value).isEmpty())` instead?

Reply:
No, .equals() can compare anything without failing, so the type of value in m.getMapName().equals(value) doesn't matter. It can't be equals() unless it is a String, but it won't cause an error if it isn't a String.